### PR TITLE
added prometheus metrics for total validators count

### PR DESCRIPTION
**Pull request by krasi-georgiev**
Signed-off-by: Krasi Georgiev <8903888+krasi-georgiev@users.noreply.github.com>

## 📝 Summary

Added some prometheus metrics

## ⛱ Motivation and Context

To allow monitoring the performance with prom and grafana and set alerts
---

## ✅ I have run these commands

* [x] `make lint`
* [x] `make test-race`
* [x] `go mod tidy`
* [x] I have seen and agree to `CONTRIBUTING.md`

### CODE REVIEW

File: .golangci.yaml [hunk 1]
```diff
@@ -76,24 +76,28 @@ linters-settings:
       #
       # Because it's easier to read without the other fields.
       #
-      - 'GetPayloadsFilters'
+      - "GetPayloadsFilters"
```
Comment (krasi-georgiev) [inline .golangci.yaml:79]: All of these are applied from the auto formatting from vscode for yuml

File: .golangci.yaml [hunk 2]
```diff
@@ -76,24 +76,28 @@ linters-settings:
...
-      - 'Webserver'
+      - "Datastore"
+      - "Housekeeper"
+      - "MockBeaconClient"
+      - "RelayAPI"
+      - "Webserver"
+
+      #
+      # Optional fields for the prometheus metrics
+      - "GaugeOpts"
```
Comment (krasi-georgiev) [inline .golangci.yaml:103]: This is the only change here, the rest is just changing from single into double quotes  applied from the auto formatting from vscode for yuml

**Comment by metachris** [comment]
interesting! what i don't like is that the datastore would need to be extended with every single prometheus metric collected. maybe better to separate the prometheus metrics into it's own component rather than adding to the datastore 🤔

**Comment by krasi-georgiev** [comment]
I am not sure what you mean, you need to have the metrics where things actually happen.

**Comment by krasi-georgiev** [comment]
just remembered that I also want to add a metric of when a validator requests a header or a payload and this one can't be in the housekeeper and needs to be in the api component

**Comment by metachris** [comment]
This is a nice PR and thanks for submitting it. However we don't want to add Prometheus metrics to the codebase at the current time. Happy to put up a link to your branch / PR, maybe from the wiki or the readme, as inspiration for others or when we revisit this in the future.

**Pull request closed without merging**